static/src/javascripts/projects/common/modules/asyncCallMerger.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+/**
+ * This module is used to merge calls to async functions that use callbacks. Ensuring the target function is
+ * called once only but the result is passed to all the call backs that require it.
+ */
+
+type Status = 'init' | 'waiting' | 'complete';
+
+type Callback = (...as: any[]) => void;
+
+type TargetCallback = (f: Callback) => void;
+
+/**
+ * Creates a function that will merge calls to the supplied target function
+ * @param {function} function to be merged, the function takes one argument, a callback that will
+ * be called with the result of the async request
+ * @return {?Date} the current date
+ */
+const mergeCalls = (targetFunction: TargetCallback): TargetCallback => {
```

Comment:
I guess we can kill these JSDoc comments now that we have flow (also a discrepancy between comment and Flow return type?)